contracts/src/systems/summit.cairo

```diff
@@ -450,12 +528,24 @@ pub mod summit_systems {
         /// @notice this function is used to assert that a beast can consume an adventurer
         /// @param beast the beast to check
         /// @param adventurer the adventurer to check
-        fn _assert_beast_can_consume(self: @ContractState, beast: Beast, adventurer: Adventurer) {
-            let total_health = beast.stats.live.bonus_health
-                + beast.stats.fixed.starting_health.into();
-            assert(total_health <= BEAST_MAX_HEALTH, errors::BEAST_MAX_HEALTH);
+        fn _assert_beast_can_consume(
+            self: @ContractState, beast: Beast, adventurer_id: u64, adventurer: Adventurer
+        ) {
+            assert(
+                beast.stats.live.bonus_health < BEAST_MAX_BONUS_HEALTH,
+                errors::BEAST_MAX_BONUS_HEALTH
+            );
             assert(adventurer.health == 0, errors::ADVENTURER_ALIVE);
             assert(adventurer.rank_at_death == 0, errors::ADVENTURER_RANKED);
+            assert(
+                get!(self.world(), (adventurer_id), AdventurerConsumed).beast_token_id == 0,
+                errors::ADVENTURER_ALREADY_CONSUMED
+            );
+        }
+
+        fn _assert_beast_can_be_revived(self: @ContractState, beast: Beast, potion_count: u8) {
+            assert(beast.stats.live.current_health == 0, errors::BEAST_ALIVE);
+            assert(potion_count >= beast.stats.live.revival_count, errors::NOT_ENOUGH_CONSUMABLES);
```

Comment:
the current code could allow for a client bug in which the player uses more revival potions than necessary. To prevent this, the contract should check to ensure potion_count == revival count